### ISSUE TITLE
Understand why active learning gets worse with more data

### ISSUE THREAD

**Issue by WardLT** [exalearn/electrolyte-design#15]
We find the performance of our active learning agent gets worse as we retrain the models. The chart below shows how we find fewer high-performing molecules with strategies where we update the model (`update` and `train`) than we do with a strategy where we never update the MPNNs (`no-retrain`)

![image](https://user-images.githubusercontent.com/7003149/116586894-8bb4fe00-a8df-11eb-902e-0a71bc247520.png)

A list of hypotheses:
- [x] Models are getting worse with training. *No, the accuracy of models on a hold-out set improves with retraining*
- [x] Too much time is getting spent on retraining. *No, the decrease in number of molecules found is 57% and only we only loose 3% of simulation time to ML tasks.*
- [x] Calibration of uncertainties changes (worsens?) with retraining. *Yes. Not yet sure if fixing that would fix our issues.*

Potential solutions:
- [x] Retrain the model from scratch. *Did not work (see above picture)*
- [x] Update the models using bootstrapped datasets (we used the full training set previously). 
- [ ] Use a larger ensemble of machine learning models (https://arxiv.org/abs/1906.02530 notes ~5 should be sufficient?)
- [ ] On-line calibration (#14)
- [ ] Bayesian neural networks instead of ensembles

**Comment by pythonpanda2** [comment]
@WardLT  Something to consider going forward. https://github.com/uncertainty-toolbox/uncertainty-toolbox

**Comment by WardLT** [comment]
The de-calibration is lessened if we use bootstrap sampling when creating the training set before updating the model.

![image](https://user-images.githubusercontent.com/7003149/116606770-d0985f00-a8f6-11eb-8815-f0a8225851b6.png)

**Comment by WardLT** [comment]
We see similar, slight degradation with the 16 bootstraped models
![image](https://user-images.githubusercontent.com/7003149/116618417-a77ecb00-a904-11eb-9997-a4a888ab04d6.png)

**Comment by WardLT** [comment]
Training with more epochs (here, 512) can make the problem worse

![image](https://user-images.githubusercontent.com/7003149/116723296-f8460080-a9a4-11eb-829d-e139681ceaa6.png)

**Comment by WardLT** [comment]
Resetting the weights on the optimizer does seem to help. This is back to using 64 epochs to retrain the model.

Using random initial weights seems to work just as well in terms of the uncertainties.

![image](https://user-images.githubusercontent.com/7003149/116731438-4d870f80-a9af-11eb-86b1-aef89a359078.png)

**Comment by WardLT** [comment]
It was a bug 😆 5e4a8b2
